File: sigma_marketing_api/SigmaMarketing/SigmaMarketing.Core/Entities/AchievementType.cs
using SigmaMarketing.Core.Entities.Base;

namespace SigmaMarketing.Core.Entities
{
    public class AchievementType : BaseEntity
    {
        public int Type { get; set; }
        public string ImageUrl { get; set; }
        public string Name { get; set; }
        public string Explanation { get; set; }
        public int Value { get; set; }
    }
}

File: sigma_marketing_app/sigma_marketing/lib/data/models/response/campaign/achievement_type.dart
import 'package:json_annotation/json_annotation.dart';

part 'achievement_type.g.dart';

@JsonSerializable()
class AchievementType {
  @JsonKey(name: 'id')
  int id = 0;

  @JsonKey(name: 'name')
  String name = "";

  @JsonKey(name: 'explanation')
  String explanation = "";

  @JsonKey(name: 'imageUrl')
  String imageUrl = "";

  @JsonKey(name: 'value')
  int value = 0;

  @JsonKey(name: 'valueCash')
  double valueCash = 0.0;

  @JsonKey(name: 'type')
  int type = 0;

  AchievementType({
    required this.id,
    required this.name,
    required this.explanation,
    required this.imageUrl,
    required this.value,
    required this.valueCash,
    required this.type,
  });

  factory AchievementType.fromJson(dynamic json) =>
      _$AchievementTypeFromJson(json);

  Map<String, dynamic> toJson() => _$AchievementTypeToJson(this);
}